code/modules/mob/living/basic/space_fauna/paper_wizard/paper_wizard.dm

```diff
@@ -0,0 +1,189 @@
+/mob/living/basic/paper_wizard
+	name = "Mjor the Creative"
+	desc = "A wizard with a taste for the arts."
+	mob_biotypes = MOB_ORGANIC|MOB_HUMANOID
+	faction = list(FACTION_HOSTILE, FACTION_STICKMAN)
+	icon = 'icons/mob/simple/simple_human.dmi'
+	icon_state = "paperwizard"
+	gender = MALE
+
+	response_help_continuous = "brushes"
+	response_help_simple = "brush"
+	response_disarm_continuous = "pushes"
+	response_disarm_simple = "push"
+
+	maxHealth = 1000
+	health = 1000
+	melee_damage_lower = 10
+	melee_damage_upper = 20
+	obj_damage = 50
+	attack_sound = 'sound/hallucinations/growl1.ogg'
+	ai_controller = /datum/ai_controller/basic_controller/paper_wizard
+	///the list of our clones
+	var/list/copies = list()
+	///spell to summon minions
+	var/datum/action/cooldown/spell/wizard_summon_minions/summon
+	///spell to summon clones
+	var/datum/action/cooldown/spell/pointed/wizard_mimic/mimic
+
+
+/mob/living/basic/paper_wizard/Initialize(mapload)
+	. = ..()
+	apply_dynamic_human_appearance(src, mob_spawn_path = /obj/effect/mob_spawn/corpse/human/wizard/paper)
+	summon = new(src)
+	summon.Grant(src)
+	ai_controller.set_blackboard_key(BB_WIZARD_SUMMON_MINIONS, summon)
+	mimic = new(src)
+	mimic.Grant(src)
+	ai_controller.set_blackboard_key(BB_WIZARD_MIMICS, mimic)
+	RegisterSignal(src, COMSIG_MOVABLE_PRE_MOVE, PROC_REF(animate_step))
+	RegisterSignal(src, COMSIG_LIVING_HEALTH_UPDATE, PROC_REF(on_health_change))
```

Comment:
this is likely my bad since back in #76570 I advised them to use Signals for stuff since we could inject at precise points in the proc rather than have stuff potentially break if something gets changed upstream on the parent proc (or someone injects new behavior on the `/basic` level or something stupid like that. do you not think it's a better way to go? extending the proc seems simple enough but I think everything could be slightly more precise using signals in the logic rather than having to overwrite parent/rely on parent.

Reply:
If there's a legit reason like proc ordering I can understand using a signal but for, say, `COMSIG_LIVING_HEALTH_UPDATE` - the signal's already sent at the end of the proc, there's no real reason to use a signal over just extending the proc

Reply:
do i use the signels or no